src/components/structures/MatrixChat.tsx

```diff
@@ -394,7 +402,8 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
         }
 
         const crossSigningIsSetUp = cli.getStoredCrossSigningForUser(cli.getUserId());
-        if (crossSigningIsSetUp) {
+        const crossSignInVerifySkip = SettingsStore.getValue(UIFeature.SkipPreloginSessionVerifyPrompt);
```

Comment:
Rather than having a negative name for config items, we are trying to have them positive, so should SkipPreloginSessionVerifyPrompt be called ShowPreloginSessionVerifyPrompt?  And the local variable crossSignInVerifySkip be called showPreloginSessionVerifyPrompt

Reply:
Your config value is postfixed with Prompt, so in my opinion, you are controlling if you are showing that Prompt.  You do raise a good point, that just because the prompt is hiding what should the functional behaviour be ... always perform the session verification or do not perform the session verification.  So I'd suggest a config value name of PerformSessionVerification ... I'm not sure about the prelogin part of the name as the user is already authenticated/logged in by this point.

Reply:
Action has been taken everything is renamed to PerformSessionVerification. At the moment even when front end refers to url for request which is "chs..." how is it happening I do not know. I have temporarily used front end switch until I find fix for this.

Reply:
Please refer to pull request #11 for all new changes as git has closed this pull request automatically somehow and ability to redo pull request is disabled .